client/packages/system/src/Vaccination/Components/VaccinationModal.tsx

```diff
@@ -0,0 +1,220 @@
+import { VaccinationCourseDoseFragment } from '../api/operations.generated';
+
+interface VaccinationModalProps {
+  vaccinationId: string | undefined;
+  vaccineCourseDoseId: string;
+  isOpen: boolean;
+  onClose: () => void;
+  defaultClinician?: Clinician;
+}
+
+export const VaccinationModal = ({
+  isOpen,
+  onClose,
+  vaccineCourseDoseId,
+  vaccinationId,
+  defaultClinician,
+}: VaccinationModalProps) => {
+  const t = useTranslation('dispensary');
+  const { success, error } = useNotification();
+  const {
+    draft,
+    updateDraft,
+    query: { dose, isLoading },
+    isDirty,
+    isComplete,
+  } = useVaccination({ vaccineCourseDoseId, vaccinationId, defaultClinician });
+
+  const { Modal } = useDialog({ isOpen, onClose, disableBackdrop: true });
+  const height = useKeyboardHeightAdjustment(620);
+
+  const save = async () => {
+    try {
+      // TODO: actual save :)
+      success(t('messages.vaccination-saved'))();
+      onClose();
+      // }
+    } catch (e) {
+      error(t('error.failed-to-save-vaccination'))();
+      console.error(e);
+    }
+  };
+
+  const modalContent = isLoading ? (
+    <BasicSpinner />
+  ) : (
+    <VaccinationForm updateDraft={updateDraft} draft={draft} dose={dose} />
+  );
+
+  return (
+    <Modal
+      title={dose?.label ?? t('label.vaccination')}
+      cancelButton={<DialogButton variant="cancel" onClick={onClose} />}
+      okButton={
+        <DialogButton
+          disabled={!isDirty || !isComplete}
+          variant="ok"
+          onClick={save}
+        />
+      }
+      height={height}
+      width={550}
+      slideAnimation={false}
+    >
+      {modalContent}
+    </Modal>
+  );
+};
+
+const VaccinationForm = ({
+  draft,
+  dose,
+  updateDraft,
+}: {
+  dose?: VaccinationCourseDoseFragment;
+  draft: VaccinationDraft;
+  updateDraft: (update: Partial<VaccinationDraft>) => void;
+}) => {
+  const t = useTranslation('dispensary');
+
+  const vaccineItemOptions = useMemo(() => {
+    return (
+      dose?.vaccineCourse.vaccineCourseItems?.map(item => ({
+        label: item.name,
+        value: item.id,
+      })) ?? []
+    );
+  }, [dose?.id]);
+
+  if (!dose) {
+    return null;
+  }
+
+  return (
+    <Container
+      maxWidth="xs"
+      sx={{ display: 'flex', flexDirection: 'column', gap: 2 }}
+    >
+      <InputWithLabelRow
+        label={t('label.clinician')}
+        Input={
+          <Grid item flex={1}>
+            <ClinicianSearchInput
+              onChange={clinician => {
+                updateDraft({
+                  clinician: clinician?.value,
+                });
+              }}
+              clinicianValue={draft.clinician}
+            />
+          </Grid>
+        }
+      />
+      <InputWithLabelRow
+        label={t('label.date')}
+        Input={
+          <DatePicker
+            disableFuture
+            value={draft.date}
+            onChange={date => updateDraft({ date })}
+            sx={{ flex: 1 }}
+          />
```

Comment:
Question: do we plan to restrict the minimum date to the minimum age for the dose? Would it be hard to do now, or leave till later?

Reply:
Thanks for spotting this! I checked with Mark - they'd like to leave it up to the clinician's discretion when they administer the dose 🙏